### ISSUE TITLE
Zip error launching newly copied resource into JupyterHub

### ISSUE THREAD

**Issue by devincowan** [hydroshare/hydroshare#5128]
**Description of the bug**
Related to #5076 
I'm opening this issue on Hydroshare repo instead of [nbfetch](https://github.com/hydroshare/nbfetch) so that we don't loose track of it...

![image](https://github.com/hydroshare/hydroshare/assets/17934193/1368c9e2-9956-4c66-a38e-52bc27634dcd)

```
Successfully established a connection with HydroShareTraceback (most recent call last):
File "/opt/conda/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
File "/opt/conda/lib/python3.9/threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
File "/opt/conda/lib/python3.9/site-packages/nbfetch/handlers.py", line 220, in pull
    raise e
File "/opt/conda/lib/python3.9/site-packages/nbfetch/handlers.py", line 214, in pull
    for line in hs.pull():
File "/opt/conda/lib/python3.9/site-packages/nbfetch/pull.py", line 59, in pull
    with zipfile.ZipFile(downloaded_zip, 'r') as zip_ref:
File "/opt/conda/lib/python3.9/zipfile.py", line 1257, in __init__
    self._RealGetContents()
File "/opt/conda/lib/python3.9/zipfile.py", line 1324, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

I have not troubleshooted further to determine what is particular to this resource that causes this issue. My guess is that it has something to do with the directory structure in the resource...

Steps to reproduce the bug:
1. Copy this resource: https://www.hydroshare.org/resource/344350caa2e341f5a8a9eb33cbed3f78/
2. Open with CUAHSI JupyterHub
3. See zip error
4. Go back to HS and "Open with CUAHSI JupyterHub" again
5. JH successfully opens the resource

**Expected behavior**
JH should successfully unpack the resource on the first try

**Additional information**
HS v2.6.1
Nbfetch: present in https://github.com/hydroshare/nbfetch/commit/2c203a57a7466f7f4011b5fe4c4be1e3fd786fc9 as well as https://github.com/hydroshare/nbfetch/tree/56429ba1063327992074ddb952aba1c042cf9575 and all other versions that I tested

**Comment by devincowan** [comment]
seems like this might be dependent on resource size
I'm able to reproduce locally for resources larger than ~200mb

**Comment by devincowan** [comment]
released as 2.7.1
